src/ucp/proto/proto_init.c

```diff
@@ -451,8 +390,16 @@ ucp_proto_common_init_xfer_perf(const ucp_proto_common_init_params_t *params,
         }
     } else if (!(params->flags & UCP_PROTO_COMMON_INIT_FLAG_RKEY_PTR)) {
         ucs_assert(reg_md_map == 0);
+
+        if ((params->flags & UCP_PROTO_COMMON_INIT_FLAG_REMOTE_ACCESS) &&
+            (params->send_mem_type != UCS_MEMORY_TYPE_UNKNOWN)) {
+            remote_mem_type = params->send_mem_type;
+        } else {
+            remote_mem_type = UCS_MEMORY_TYPE_HOST;
```

Comment:
can we always use  params->send_mem_type without extra conditions? and initialize the protocol parameters accordingly?